tests/cli/test_cli.py

```diff
@@ -0,0 +1,169 @@
+import os
+from shutil import rmtree
+
+import pytest
+from click.testing import CliRunner
+
+from sqlmesh.cli.example_project import init_example_project
+from sqlmesh.cli.main import cli
+
+
+@pytest.fixture(scope="session")
+def runner() -> CliRunner:
+    return CliRunner()
+
+
+def create_example_project(temp_dir) -> None:
+    """
+    Sets up CLI tests requiring a real SQLMesh project by:
+        - Creating the SQLMesh example project in the temp_dir directory
+        - Overwriting the config.yaml file so the duckdb database file will be created in the temp_dir directory
+    """
+    init_example_project(temp_dir, "duckdb")
+    with open(temp_dir / "config.yaml", "w") as f:
+        f.write(
+            f"""gateways:
+    local:
+        connection:
+            type: duckdb
+            database: {temp_dir}/db.db
+
+default_gateway: local
+
+model_defaults:
+    dialect: duckdb
+"""
+        )
+
+
+def reset_example_project(temp_dir) -> None:
```

Comment:
I wonder if we should just split each "reset" into a separate test. It would result in specific commands having multiple tests but I would prefer that over having to reset state between tests and potentially have that shared state, if not properly reset, affect the results.